File: week1/f.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;
stack <char> s;
int main()
{
	char c; 
    bool result = true; 
	string str; 
    cin >> str;
	for (int i = 0; i < str.size(); i++) { 
		c = str[i];
		if (c == '(' || c == '{' || c == '[') {
			s.push(c); 
            continue;
		}						
		switch (c){	
		case ')': if (!s.empty() && s.top() == '(') s.pop();
			else result = false; 
			break;
		case '}': if (!s.empty() && s.top() == '{') s.pop(); 
			else result = false; 
			break;
		case ']': if (!s.empty() && s.top() == '[') s.pop(); 
			else result = false; 
			break;
		}
	}
    if(result== true && s.empty()){
        cout << "yes";
    }
    else{
        cout << "no";
    }
	return 0;
}

File: week1/f.py
N = input()
print (N)
 
M={'(':')','[':']','{':'}'}
stack=[]
flag=0
 
for i in N:
    if i=='(' or i=='[' or i=='{': stack.append(i)
    if (i==')' or i==']' or i=='}') and (len(stack)==0 or i!=M[stack[(len(stack)-1)]]):
        flag=1
        break
    if (i==')' or i==']' or i=='}') and i==M[stack[(len(stack)-1)]]:
        stack.pop()
        continue
    
 
if flag==1 or len(stack)>0:
    print ("Неправильная последовательность скобок")
else:
    print ("Правильная последовательность скобок")
